src/theory/fp/theory_fp.cpp

```diff
@@ -739,6 +767,13 @@ void TheoryFp::postCheck(Effort level)
 bool TheoryFp::preNotifyFact(
     TNode atom, bool pol, TNode fact, bool isPrereg, bool isInternal)
 {
+  /* Word-blast lazier if configured. */
+  if (options::fpLazyWb() && d_wbFactsCache.find(fact) == d_wbFactsCache.end())
+  {
+    d_wbFactsCache.insert(fact);
+    convertAndEquateTerm(fact);
```

Comment:
Couldn't we just pass `atom` to `convertAndEquateTerm()` here and avoid the `if (options::fpLazyWb() && node.getKind() == kind::NOT)` block above? I think that this would also make the caching a bit more effective because right now, if we get `(not l)` and later `l` (or vice-versa), we will attempt to word-blast `l` twice.

Reply:
Good point. Updated and currently running on the cluster.